frontend/kendra-button-front/src/components/SignUp.tsx

```diff
@@ -0,0 +1,154 @@
+import { AmplifyButton, AmplifyFormField, AmplifyPasswordField } from '@aws-amplify/ui-react';
+import { ReactElement, useRef, useState, Dispatch, SetStateAction } from 'react'
+import { Auth } from 'aws-amplify';
+import { AuthState } from '@aws-amplify/ui-components';
+
+interface Props {
+  setScreen?: Dispatch<SetStateAction<string>>;
+}
+
+const SignUp = (props: Props): ReactElement => {
+  const { setScreen } = props;
+  const email = useRef('');
+  const password = useRef('');
+  const [displayAcc, setDisplayAcc] = useState<boolean>(false);
+  const [signupAccSuccess, setSignupAccSuccess] = useState<string>('');
+  const [signupAccErr, setSignupAccErr] = useState<string>('');
+
+
+  const toggleDisplay = (): void => {
+    setDisplayAcc(!displayAcc);
+    setSignupAccErr('');
+    setSignupAccSuccess('');
+  };
+
+
+  const onEmailChange = (e): void => {
+    email.current = e.target.value;
+  };
+
+  const onPasswordChange = (e): void => {
+    password.current = e.target.value;
+  };
+
+  const toSignIn = () => {
+    setScreen(AuthState.SignIn)
+  }
+
+  const onSubmit = async (e): Promise<void> => {
+    let errors = []
+    if (email.current.length === 0) {
+      errors.push('Please enter your email');
+    }
+    if (password.current.length === 0) {
+      errors.push('Please enter your password');
+    }
+    if (errors.length > 0) {
+      setSignupAccErr(errors.join(' / '));
+      return;
+    }
+
+    try {
+      const res = await Auth.signUp(
+        email.current,
+        password.current
+      );
+      if (res) {
+        setSignupAccErr('');
+        setSignupAccSuccess('Sign Up Completed!')
+      }
+
+    } catch (e) {
+      console.log('error e ', e);
+      let errormsg;
+      if (e.code === 'InvalidParameterException') {
+        let errors = [];
+        if (e.message.indexOf('password') > -1) {
+          errors.push('Password must contain the following: \n - Minimum length, which must be at least 6 characters but fewer than 99 characters. \n - Require numbers. \n - Require uppercase letters. \n - Require lowercase letters.')
+        }
+        errormsg = errors.join('\n');
+      }
+      setSignupAccErr(errormsg);
+    }
+    return;
+  };
+
+  return (
+    <>
+      <div
+        className={`align-items-center d-flex justify-content-between pb-1 p-3`}
+      >
+        <div className="card w-100">
```

Comment:
Form을 보기에 너무 폭이 좁아 보이고, 모바일 대응도 되었으면 하는 바램에서 이렇게 제안 해 봅니다

```suggestion
        <div className="card col-sm-6 h-75 jsx-466828807 overflow-auto p-0 justify-content-between">
```
- `col-sm-6`: SignUp Form 화면이 너무 좁게 보여서 화면의 절반 크기로만 보이면 좋을 것 같아서 bootstrap 클래스를 넣었고, 이 **sm** 클래스를 적용하면 width가 576px 이상일 때에만 화면 크기가 절반으로 보이고, 그 미만이면 width가 100%로 보이게 됩니다. 해당 문서를 보시면 도움이 되실겁니다 [Bootstrap: Grid options](https://getbootstrap.com/docs/4.0/layout/grid/#grid-options)
- `h-75, overflow-auto`: 양식 컴포넌트의 최대 height가 75%로 맞춰져서 만약 그 안의 내용이 해당 height를 넘어가면 overflow-auto가 먹어서 자동으로 스크롤이 생기게 됩니다
- `p-0`: **card** 클래스가 기본적으로 padding 이 먹기 때문에 지금 컴포넌트에서는 이쁘게 출력되지 않게 때문에 padding을 없앴습니다
- `justify-content-between`: **h-75** 클래스 때문에 세로로 길어졌으므로 내용을 가운데에 보이게 하기 위함입니다. 여기서는 만약에 내용이 길어졌을 경우 상단 내용이 잘려서 보이는 현상이 발생되므로, 편법으로 알맹이 내용 상단과 하단에 empty한 <div>를 넣어서 내용이 모두 잘 보이도록 하는 게 좋겠습니다

Reply:
제안 해주신대로 수정했습니다!
p-3 으로 수정했습니다.